src/python/grpcio/grpc/_cython/_cygrpc/aio/call.pyx.pxi

```diff
@@ -78,17 +77,21 @@ cdef class _AioCall:
         """Destroys the corresponding Core object for this RPC."""
         grpc_call_unref(self._grpc_call_wrapper.call)
 
-    cdef AioRpcStatus _cancel_and_create_status(self, object cancellation_future):
-        """Cancels the RPC in Core, and return the final RPC status."""
-        cdef AioRpcStatus status
+    def cancel(self, AioRpcStatus status):
+        """Cancels the RPC in Core with given RPC status.
+        
+        Above abstractions must invoke this method to set Core objects into
+        proper state.
+        """
```

Comment:
What would happen if we call the `cancel` method before the `channel.unary_unary` gets scheduled by the `UnaryUnaryCall`? Should we prevent to not call `grpc_call_cancel_with_status` when the call has been never started?

Reply:
In unit test like `test_cancel_unary_unary` or `test_early_cancel_unary_stream`, the underlying AsyncIO task is created, but its function statement won't be executed. So, those unit tests are testing the scenario of calling `grpc_call_cancel_with_status` early, and the result seems safe.